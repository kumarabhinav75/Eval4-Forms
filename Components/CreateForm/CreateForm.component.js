import React, { Component } from 'react';
import {
  TextInput, TouchableOpacity, Text, ScrollView, View,
} from 'react-native';
import createFormAPI from '../../src/helpers/createForm';

import styles from './CreateForm.style';

export default class CreateForm extends Component {
  state = {
    formname: '',
    fields: [],
    fieldCount: 0,
  }

  handleFormNameChange = (formname) => {
    this.setState({
      formname,
    });
  }

  addNewField = () => {
    const newFieldCount = this.state.fieldCount + 1;
    this.setState({
      fieldCount: newFieldCount,
    });
  }

  handleFieldTextEnd = (fieldName) => {
    const fieldArr = this.state.fields;
    fieldArr.push(fieldName);
    this.setState({
      fields: fieldArr,
    });
  }

  saveForm = () => {
    // console.log('HELLO');
    const { fields, formname } = this.state;
    const fieldObject = {};
    fields.map((fieldName) => {
      fieldObject[fieldName] = [];
    });
    const formObj = {};
    formObj.formname = formname;
    formObj.field = { ...fieldObject };
    createFormAPI(formObj);
    this.props.navigation.navigate('MainScreen');
  }

  displayNewField = count => Array(count).fill(
    <TextInput
      placeholder="Field"
      returnKeyType="next"
      keyboardType="email-address"
      style={styles.input}
      onEndEditing={e => this.handleFieldTextEnd(e.nativeEvent.text)}
    />,
  )

  render() {
    console.log(this.state.fields);
    // moment().format('hh:mm:ss a');
    return (
      <View style={styles.container}>
        <TextInput
          placeholder="FormName"
          value={this.state.formname}
          returnKeyType="next"
          keyboardType="email-address"
          style={styles.input}
          onChangeText={this.handleFormNameChange}
        />
        <TouchableOpacity style={styles.buttonContainer} onPress={this.addNewField}>
          <Text style={styles.buttonText}>Add Field</Text>
        </TouchableOpacity>
        <ScrollView>
          {this.displayNewField(this.state.fieldCount)}
        </ScrollView>
        <TouchableOpacity style={styles.saveButtonContainer} onPress={this.saveForm}>
          <Text style={styles.buttonText}>Save</Text>
        </TouchableOpacity>
      </View>
    );
  }
}
