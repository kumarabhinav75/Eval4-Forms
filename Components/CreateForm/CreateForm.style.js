import {StyleSheet} from 'react-native';
export default styles = StyleSheet.create({  
  input: {
    height: 40,
    backgroundColor: 'rgba(255,255,255,0.2)',
    marginBottom: 10,
    color: '#FFF',
    paddingHorizontal: 10,
  },
  container: {
    flex: 1,
    // justifyContent: 'center',
    // alignItems: 'center',
    backgroundColor: 'orange',
  },
  buttonContainer: {
    backgroundColor: '#2980b9',
    paddingVertical: 15,
    width: 70,
    left: 300,
  },
  buttonText: {
    textAlign: 'center',
    color: '#FFFFFF',
    fontWeight: '700',
  },
})