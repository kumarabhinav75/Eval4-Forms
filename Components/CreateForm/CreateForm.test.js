import React from 'react';
import renderer from 'react-test-renderer';
import CreateForm from './CreateForm.component';

describe(CreateForm, () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<CreateForm />);
    expect(tree).toMatchSnapshot();
  });
});
