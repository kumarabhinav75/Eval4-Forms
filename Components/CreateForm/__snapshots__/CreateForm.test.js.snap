// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateForm should match the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "orange",
      "flex": 1,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    keyboardType="email-address"
    onChangeText={[Function]}
    placeholder="FormName"
    returnKeyType="next"
    style={
      Object {
        "backgroundColor": "rgba(255,255,255,0.2)",
        "color": "#FFF",
        "height": 40,
        "marginBottom": 10,
        "paddingHorizontal": 10,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#2980b9",
        "left": 300,
        "opacity": 1,
        "paddingVertical": 15,
        "width": 70,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Add Field
    </Text>
  </View>
  <RCTScrollView>
    <View />
  </RCTScrollView>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#2980b9",
        "bottom": 0,
        "opacity": 1,
        "paddingVertical": 15,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Save
    </Text>
  </View>
</View>
`;
